src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -173,11 +169,19 @@ async Task AfterProvisionAsync(IAzureResource resource)
         // If we created any resources then save the user secrets
         if (userSecretsPath is not null)
         {
-            // Ensure directory exists before attempting to create secrets file
-            Directory.CreateDirectory(Path.GetDirectoryName(userSecretsPath)!);
-            await File.WriteAllTextAsync(userSecretsPath, userSecrets.ToString(), cancellationToken).ConfigureAwait(false);
+            try
+            {
+                var userSecrets = await provisioningContextLazy.Value.ConfigureAwait(false);
```

Comment:
I am very confused by this. Shouldn't this be the provisioningContext?
```suggestion
                var provisioningContext = await provisioningContextLazy.Value.ConfigureAwait(false);
```

Reply:
Fixed. That was a bit of a brainfart. I didn't catch it because everything can ToString :)